### PR TITLE
Changes tree mutations to use infusions

### PR DESCRIPTION
**Pull request by TobleroneSwordfish**
[FEATURE] [HYDROPONICS]
## About the PR 
Changes tree mutations to (mostly) use infusions rather than random chance.
- Paper tree now requires infusion with paper
- Money trees now only mutate from paper trees with >= 30 potency
- Dogwood tree now requires infusion with aconitine
- Rubber tree now requires infusion with liquid rubber
- Sassafras has been left as a random mutation both because I couldn't think of a chem and because it was already base chance to occur.

Since liquid rubber was kind of hard to get previously, it can now be made by exposing any rubber item (not cables) to a temperature of 180C, causing it to melt into a puddle of 10u of rubber.
I'm reasonably sure this won't break anything horribly, but if it does I can change it to be an acetone interaction or something.

## Why's this needed? 
Trees have a lot of mutations, getting any specific mutation was an RNG mess, this should make it a lot more enjoyable.

## Changelog

```changelog
(u)LeahTheTech
(*)Most tree mutations now require reagent infusions.
(*)Rubber items now melt into liquid rubber above 180C.
```

### CODE REVIEW

**Comment by Momoberry** [comment]
Great change, the botany waiting game is killer and the seed spam strat is awful gameplay.

**Comment by Munien7** [comment]
Was actually meaning to do this myself at some point. Anything that removes the annoying rng seed spam especially with trees is a godsend.

File: code/modules/materials/Mat_MaterialProcs.dm [hunk 1]
```diff
@@ -680,3 +680,14 @@ triggerOnEntered(var/atom/owner, var/atom/entering)
...
+	execute(var/atom/location, var/temp)
+		if (istype(owner.owner, /obj/cable) || istype(owner.owner, /obj/item/cable_coil)) //let's just not deal with that
+			return
+		if (temp >= T0C + 180) //wikipedia told me rubber melts at 180C
+			var/datum/reagents/temp_reagents = new(10)
+			temp_reagents.add_reagent("rubber", 10)
+			temp_reagents.set_reagent_temp(temp)
+			temp_reagents.reaction(get_turf(owner.owner))
+			qdel(temp_reagents)
+			qdel(owner.owner)
```
Comment (Tarmunora) [inline code/modules/materials/Mat_MaterialProcs.dm:693]: how does this work with e.g.: synthrubber turfs?

Comment (TobleroneSwordfish) [reply code/modules/materials/Mat_MaterialProcs.dm:693]: I can't get the `materialProc` to trigger for a turf at all so I assume it just doesn't. I'll add a check to make sure we never `qdel` a turf anyway.

Comment (Tarmunora) [reply code/modules/materials/Mat_MaterialProcs.dm]: 👍

**Comment by Sbmhawk** [comment]
I thought you needed rubber trees to get liquid rubber, what's the other method?

**Comment by TobleroneSwordfish** [comment]
> I thought you needed rubber trees to get liquid rubber, what's the other method?

See PR body: rubber items now melt

**Comment by Sbmhawk** [comment]
Oh, I somehow missed that lmao

**Pull request merged**